generator_cargohome.go

```diff
@@ -32,25 +33,215 @@ func (s *Spec) HasCargohomes() bool {
 
 func withCargohome(g *SourceGenerator, srcSt, worker llb.State, opts ...llb.ConstraintsOpt) func(llb.State) llb.State {
 	return func(in llb.State) llb.State {
-		workDir := "/work/src"
-		joinedWorkDir := filepath.Join(workDir, g.Subpath)
+		// Extract platform information from constraints
+		var constraints llb.Constraints
+		for _, opt := range opts {
+			opt.SetConstraintsOption(&constraints)
+		}
+
+		// Determine if this is a Windows platform
+		isWindows := constraints.Platform != nil && constraints.Platform.OS == "windows"
+
+		// Set up paths and scripts based on platform
+		var (
+			workDir, joinedWorkDir string
+			sccachePath            string
+			sccacheInstallScript   []byte
+			scriptName             string
+			shellCommand           string
+		)
+
+		if isWindows {
+			workDir = "C:\\work\\src"
+			joinedWorkDir = filepath.Join(workDir, g.Subpath)
+			sccachePath = "C:\\temp\\dalec\\sccache-binary-cache"
+			scriptName = "install_sccache.ps1"
+			shellCommand = "powershell -File C:\\temp\\install_sccache.ps1"
+
+			sccacheInstallScript = []byte(`# PowerShell script to install sccache
```

Comment:
Powershell? Wha?


In any case we can download/verify sccache with `llb.HTTP` (or even use an dalec.SourceHTTP)

Reply:
Note taht sccache needs to be in the build environment and not in the generator.